services/src/main/java/org/openlearn/service/PortfolioItemService.java

```diff
@@ -130,21 +138,26 @@ public PortfolioItemDTO findOne(final Long id) {
 	public void delete(final Long id) {
 		log.debug("Request to delete portfolio item : {}", id);
 		PortfolioItem portfolioItem = portfolioItemRepository.findOne(id);
-		if (portfolioItem != null && (SecurityUtils.isAdmin() || inOrgOfCurrentUser(portfolioItem))) {
+		if (portfolioItem == null) new PortfolioItemNotFoundException(id);
+
+		if ((SecurityUtils.isAdmin() || inOrgOfCurrentUser(portfolioItem))) {
+			fileInformationService.deleteByPortfolioItem(portfolioItem);
 			portfolioItemRepository.delete(id);
 		} else {
-			// TODO: Error handling / logging
+			throw new AccessDeniedException();
 		}
 	}
 
-	private boolean inOrgOfCurrentUser(final PortfolioItemDTO portfolioItemDTO) {
+	public boolean inOrgOfCurrentUser(final PortfolioItemDTO portfolioItemDTO) {
 		User user = userService.getCurrentUser();
 		User student = userRepository.findOneByIdAndAuthority(portfolioItemDTO.getStudentId(), STUDENT);
-		return student != null && user.getOrganization().equals(student.getOrganization());
+		return student != null && SecurityUtils.isCurrentUserInRole(AuthoritiesConstants.ORG_ADMIN) &&
```

Comment:
Yeah, I think the permission structure will change here anyway. At the moment that was added to avoid allowing instructors or students in the same organization from having access to actions (like `delete`) that they shouldn't. However, it's all very messy and repetitive. I'm thinking of abstracting the permissions elsewhere into a similar sort of design in Ruby on Rails most popular authorization library [`cancancan`](https://github.com/CanCanCommunity/cancancan). That design is very DRY, and reads well.